### PR TITLE
Add additional index checks to COLLADA importer

### PR DESCRIPTION
**Pull request by W4RH4WK**
ref #46548

### CODE REVIEW

File: editor/import/editor_import_collada.cpp [hunk 1]
```diff
@@ -681,7 +681,8 @@ Error ColladaImport::_create_mesh_surfaces(bool p_optimize, Ref<EditorSceneImpor
 
 				int vertex_index = p.indices[src + vertex_ofs]; //used for index field (later used by controllers)
 				int vertex_pos = (vertex_src->stride ? vertex_src->stride : 3) * vertex_index;
-				ERR_FAIL_INDEX_V(vertex_pos, vertex_src->array.size(), ERR_INVALID_DATA);
+				ERR_FAIL_INDEX_V(vertex_pos + 0, vertex_src->array.size(), ERR_INVALID_DATA);
+				ERR_FAIL_INDEX_V(vertex_pos + 2, vertex_src->array.size(), ERR_INVALID_DATA);
```
Comment (akien-mga) [inline editor/import/editor_import_collada.cpp:685]: A simpler check might be:
```suggestion
				ERR_FAIL_INDEX_V(vertex_pos, vertex_src->array.size() - 2, ERR_INVALID_DATA);
```
no?

Comment (W4RH4WK) [reply editor/import/editor_import_collada.cpp:685]: My intention was to always check the upper and lower bound when accessing a sequence of elements. Doing it for every element in the sequence doesn't scale well, and might not catch any additional errors to just checking upper+lower.

With the suggested variant a problems arises if the array size is < 2. By `size()` returning an unsigned value, an underflow can occur when subtracting a value. A out-of-bounds access might therefore not be caught.

**Review by Calinou — APPROVED** [review]

**Comment by fire** [comment]
Having more checks doesn't affect runtime performance because the collada importer convert to the Godot internal format tscn. 

The change isn't noticeable and it shouldn't be noticeable.

👍  This pr is fine. (I did a textual review and didn't execute the project.)

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.

**Pull request merged**